### ISSUE TITLE
Sort the text detected on basis of order of appearance

### ISSUE THREAD

**Issue by kalai2033** [clovaai/CRAFT-pytorch#40]
First of all thanks very much for your model. I have used your pretrained model for text extraction. I have got the bounding box coordinates and converted the polygons to rectangles for cropping the text area in the image. The resulted bounding boxes are shuffled and i could not sort it out. As per my understanding, the boxes are sorted on the basis of Y3. But when there is a presence of curved text in a same line like in the image below, the order gets shuffled and i need to sort it before passing it to text extraction model.

![image](https://user-images.githubusercontent.com/49688647/64231526-2af9d600-cef0-11e9-8889-43f2d5a0e908.png)

In this case the bb coordinates with detected text are 

146,36,354,34,354,82,146,84    "Australian"

273,78,434,151,411,201,250,129       "Collection"

146,97,250,97,250,150,146,150       "vine"

77,166,131,126,154,158,99,197      "Old"

242,215,361,241,354,273,235,248     "Valley"

140,247,224,219,234,250,150,277    "Eden"

194,298,306,296,307,324,194,325     "Shiraz"

232,406,363,402,364,421,233,426     "Vintage"

152,402,216,405,215,425,151,422    "2008"

124,470,209,480,207,500,122,490     "South"

227,481,387,472,389,494,228,503     "Australia"

222,562,312,564,311,585,222,583     "Gibson"

198,564,217,564,217,584,198,584     "by"

386,570,421,570,421,600,386,600      "750 ml"

But the expected output is Australian->old->vine->collection->Eden->Valley->shiraz->2008->vintage->south->Australia->by->GIBSON->750ml.

**Comment by itsmenmd** [comment]
> First of all thanks very much for your model. I have used your pretrained model for text detection. I have got the bounding box coordinates as txt files and converted the polygons to rectangles for cropping the text area in the image. The resulted bounding boxes are shuffled and i could not sort it out. When there is a presence of curved text in a same line like in the image below, the order gets shuffled and i need to sort it before passing it to text extraction model.
> 
> ![image](https://user-images.githubusercontent.com/49688647/64243046-f47b8580-cf06-11e9-975b-d3e0e9afca39.png)
> 
> Converted the polygons to rectangles for cropping text areas
> 
> ![image](https://user-images.githubusercontent.com/49688647/64231526-2af9d600-cef0-11e9-8889-43f2d5a0e908.png)
> 
> I have not used POLY mode. For the above image, the model outputs a txt file in which the bb coordinates are as follows. I have added the detected text for better explanation of my problem. In this case the detection order is
> 
> 146,36,354,34,354,82,146,84 "Australian"
> 
> 273,78,434,151,411,201,250,129 "Collection"
> 
> 146,97,250,97,250,150,146,150 "vine"
> 
> 77,166,131,126,154,158,99,197 "Old"
> 
> 242,215,361,241,354,273,235,248 "Valley"
> 
> 140,247,224,219,234,250,150,277 "Eden"
> 
> 194,298,306,296,307,324,194,325 "Shiraz"
> 
> 232,406,363,402,364,421,233,426 "Vintage"
> 
> 152,402,216,405,215,425,151,422 "2008"
> 
> 124,470,209,480,207,500,122,490 "South"
> 
> 227,481,387,472,389,494,228,503 "Australia"
> 
> 222,562,312,564,311,585,222,583 "Gibson"
> 
> 198,564,217,564,217,584,198,584 "by"
> 
> 386,570,421,570,421,600,386,600 "750 ml"
> 
> But the expected output is Australian->old->vine->collection->Eden->Valley->shiraz->2008->vintage->south->Australia->by->GIBSON->750ml.

How to get text after coordinates ?

**Comment by shekarneo** [comment]
Hi if you got the solution to sort text boxes please share
